src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -261,7 +261,15 @@ public CommitLogPosition add(Mutation mutation) throws CDCWriteException
     {
         assert mutation != null;
 
-        mutation.validateSize(MessagingService.current_version, ENTRY_OVERHEAD_SIZE);
+        try
+        {
+            mutation.validateSize(MessagingService.current_version, ENTRY_OVERHEAD_SIZE);
+        }
+        catch (MutationExceededMaxSizeException e)
+        {
+            metrics.oversizedMutations.mark();
```

Comment:
can we increment this metric inside of `mutation.validateSize` so its captured in the BlockingReadRepairs mutation case earlier as well? Also then we dont need to catch and rethrow the exception.